dotnet-test-samples/async-architectures-lambda-sqs/Readme.md

```diff
@@ -0,0 +1,101 @@
+
+![Generic Asynchronous System](./img/generic.png)
+
+## Establish Logical Boundaries
+Asynchronous systems rarely exist in isolation. Typically a production system will be made up of many interconnected subsystems. In order to create a reasonable testing strategy, it is useful to break complex systems into a set of logical subsystems. A subsystem may be a group of services that work together to accomplish a single task. A subsystem should have well understood inputs and outputs. A subsystem should be small enough to be able to reason about and understand. Breaking your complex architecture into smaller subsystems makes it easier to create isolated and targeted tests.
+
+## Create Test Harnesses
+When testing asynchronous systems, it can be useful to create test harnesses. These harnesses contain resources that generate inputs for your subsystem and then receive the system’s outputs. Your tests will use the harnesses to exercise the system and determine whether it is performing as expected. These harnesses are resources that are used for testing purposes only. They are not used by production features. Test harnesses are typically deployed to pre-production environments only. However, deploying test harnesses to production environments may be useful for production systems that are tolerant of test data. In these cases you may decide to perform tests in production.
+
+## Configure Event Producers and Event Listeners
+Test harnesses are usually composed of event producers and event listeners. The producers provide input to the System Under Test (SUT) and the listeners are configured to receive output. The automated tests send configured events to the producer and then poll the listener to examine the output. If the output meets expectations the test passes.
+
+![Generic Asynchronous System Test](./img/generic-async-test.png)
+
+## Define Service Level Agreements
+Although your architecture may be asynchronous, it is still useful to establish reasonable expectations about the maximum duration your system may take to process before it considered to be in a failure state. These expectations may be explicitly defined as Service Level Agreements (SLAs). When you design your tests, you may set timeouts that match your SLA’s. If the system does not return results within the timeout period it can be considered to be in violation of the SLA and the tests will fail.
+
+
+# Asynchronous Integration Test with Amazon Simple Queue Service (SQS)
+You may use a variety of resource types to create the event listener for your asynchronous system under test ([more about event listeners](https://github.com/aws-samples/serverless-test-samples/tree/main/dotnet-test-samples/async-architectures#create-event-listeners-in-test-environments)).
```

Comment:
At the end of the refactoring this folder would be removed, need to find a solution for the common artifacts and pages.
Perhaps keeping the old page is a good idea but needs to update https://github.com/aws-samples/serverless-test-samples/tree/main/dotnet-test-samples/async-architectures to point at this pattern as well